ext/ffi/ffi.g

```diff
@@ -97,6 +99,7 @@ declarations:
 				{zend_ffi_declare(name, name_len, &dcl);}
 			)*
 		)?
+		{if (!has_name && ((common_dcl.flags & (ZEND_FFI_DCL_ENUM | ZEND_FFI_DCL_STORAGE_CLASS)) == ZEND_FFI_DCL_ENUM)) zend_ffi_cleanup_dcl(&common_dcl);}
```

Comment:
This may be done in a simpler way.

```diff
diff --git a/ext/ffi/ffi.g b/ext/ffi/ffi.g
index e30d86b21c7..7d9646d2961 100644
--- a/ext/ffi/ffi.g
+++ b/ext/ffi/ffi.g
@@ -97,7 +97,10 @@ declarations:
 				initializer?
 				{zend_ffi_declare(name, name_len, &dcl);}
 			)*
-		)?
+		|
+			/* empty */
+			{if ((common_dcl.flags & (ZEND_FFI_DCL_ENUM | ZEND_FFI_DCL_STORAGE_CLASS)) == ZEND_FFI_DCL_ENUM) zend_ffi_cleanup_dcl(&common_dcl);}
+		)
 		";"
 	)*
 ;
```

Most probably the ``((common_dcl.flags & (ZEND_FFI_DCL_ENUM | ZEND_FFI_DCL_STORAGE_CLASS)) == ZEND_FFI_DCL_ENUM)`` check also may be simplified.